JS/Lesson15/goksel/script.js

```diff
@@ -103,62 +104,66 @@ const stringExercise14 =
 const string2Exercise18 = 'world';
 const string3Exercise18 = '!';
-console.log('Exercise18: ', /* Result here */);
+console.log('Exercise18: ',string1Exercise18.concat(string2Exercise18, string3Exercise18) );
 
 console.log('------------------------');
 // 19. Use the `split` and `join` methods to reverse the words in the string stringExercise28 and print the result to the console.
 const stringExercise19 = 'This is a sample sentence.';
-console.log('Exercise19: ', /* Result here */);
+console.log('Exercise19: ', stringExercise19.split(" ").reverse().join(" ") );
 
 console.log('------------------------');
 
 // 20. Use the `charAt` method to extract the first and last characters of a string and concatenate them into a new string. Then use the `toUpperCase` method to make the new string uppercase and print the result to the console.
 const stringExercise20 = 'JavaScript';
-
-console.log('Exercise20: ', /* Result here */);
+const first = stringExercise20.charAt(0);
+const last = stringExercise20.charAt(stringExercise20.length - 1);
+const result= first + last ;
+console.log('Exercise20: ', result.toUpperCase());
 
 console.log('------------------------');
 // 21. Use the `substring` method to extract the substring "quick brown fox" from the string stringExercise32 based on two indices and print it to the console.
 const stringExercise21 = 'The quick brown fox jumps over the lazy dog.';
 
-console.log('Exercise21: ', extractedSubstringExercise32);
+console.log('Exercise21: ', stringExercise21.substring(stringExercise21.indexOf("q"), stringExercise21.indexOf("jumps") -1) );
 
 console.log('------------------------');
 // 22. Use the `endsWith` method to check if a string ends with ".com" or ".org" (case-insensitive), and print the result to the console.
 const url1Exercise22 = 'https://www.example.com';
 const url2Exercise22 = 'http://www.example.org';
 
-console.log('Exercise22: URL 1 is valid:', /* Result here */);
-console.log('Exercise22: URL 2 is valid:', /* Result here */);
+console.log('Exercise22: URL 1 is valid:', url1Exercise22.endsWith(".com") || url1Exercise22.endsWith(".org"));
+console.log('Exercise22: URL 2 is valid:', url2Exercise22.endsWith(".com") || url2Exercise22.endsWith(".org"));
 
 console.log('------------------------');
 // Bonus: 23. Use the `includes` method to check if a string includes both "apple" and "banana" (case-insensitive), and print the result to the console. Use && (logical end operator) to check if both result are true: e.g. const result = condition1 && condition2;
 const string1Exercise23 = 'I like apple and Banana';
 const string2Exercise23 = 'I like apple and banana';
 const string3Exercise23 = 'I like Apple and banana';
 
-console.log('Exercise23: ', /* Result here */);
+console.log('Exercise23a: ', string1Exercise23.includes("apple") && string1Exercise23.includes("banana") );
+console.log('Exercise23b: ', string2Exercise23.includes("apple") && string2Exercise23.includes("banana") );
```

Comment:
This solution is exactly what I was going to ask for in this exercises but I made a mistake and wrote "case-insensitive". So to solve this problem correctly, you'd need to convert the string that you check to lower case first to make sure that it's in the same case with the string that you compare it to: `string1Exercise23.toLowerCase().includes("apple") ....`

Reply:
console.log('Exercise23a: ', string1Exercise23.toLowerCase().includes("apple") && string1Exercise23.toLowerCase().includes("banana") );
console.log('Exercise23b: ', string2Exercise23.toLowerCase().includes("apple") && string2Exercise23.toLowerCase().includes("banana") );
console.log('Exercise23c: ', string3Exercise23.toLowerCase().includes("apple") && string3Exercise23.toLowerCase().includes("banana") );